From a06f805605c13e1b19fc1b707a3ac9459903ec87 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Sun, 14 Mar 2021 10:06:41 -0400
Subject: [PATCH 37/41] On Samsung R vendor, there is still the ISehMiscPower
 HIDL HAL, but there is also the AOSP's AIDL HAL. It was previously assumed
 this wouldn't happen, fix this assumption

  from
  https://github.com/phhusson/platform_frameworks_base/commit/e202b51e3522636dade7a0ede8688ccd38306468
---
 ...om_android_server_power_PowerManagerService.cpp | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/services/core/jni/com_android_server_power_PowerManagerService.cpp b/services/core/jni/com_android_server_power_PowerManagerService.cpp
index 298cd8c7171..a5812379f86 100644
--- a/services/core/jni/com_android_server_power_PowerManagerService.cpp
+++ b/services/core/jni/com_android_server_power_PowerManagerService.cpp
@@ -124,6 +124,14 @@ static HalVersion connectPowerHalLocked() {
     if (!gPowerHalHidlExists && !gPowerHalAidlExists) {
         return HalVersion::NONE;
     }
+    if (
+		    gPowerHalAidl_ == nullptr &&
+		    gPowerHalHidlV1_0_ == nullptr) {
+	ALOGE("Trying to connect to Samsung Power HAL");
+        gSehMiscPower = ISehMiscPower::getService();
+        gPowerHalHidlV1_0_ = IPowerV1_0::getService("miscpower");
+	ALOGE("Got miscpower = %d, SehMiscPower = %d", gPowerHalHidlV1_0_ != nullptr ? 1 : 0, gSehMiscPower ? 1 : 0);
+    }
     if (gPowerHalAidlExists) {
         if (!gPowerHalAidl_) {
             gPowerHalAidl_ = waitForVintfService<IPowerAidl>();
@@ -135,12 +143,6 @@ static HalVersion connectPowerHalLocked() {
             gPowerHalAidlExists = false;
         }
     }
-    if (gPowerHalHidlExists && gPowerHalHidlV1_0_ == nullptr) {
-	ALOGE("Trying to connect to Samsung Power HAL");
-        gSehMiscPower = ISehMiscPower::getService();
-        gPowerHalHidlV1_0_ = IPowerV1_0::getService("miscpower");
-	ALOGE("Got miscpower = %d, SehMiscPower = %d", gPowerHalHidlV1_0_ != nullptr ? 1 : 0, gSehMiscPower ? 1 : 0);
-    }
     if (gPowerHalHidlExists && gPowerHalHidlV1_0_ == nullptr) {
         gPowerHalHidlV1_0_ = IPowerV1_0::getService();
         if (gPowerHalHidlV1_0_) {
-- 
2.30.2

